pex/platforms.py

```diff
@@ -3,11 +3,16 @@
 
 from __future__ import absolute_import
 
-from collections import namedtuple
 from textwrap import dedent
 
+from pex.typing import TYPE_CHECKING
 
-class Platform(namedtuple("Platform", ["platform", "impl", "version", "abi"])):
+if TYPE_CHECKING:
+    from typing import Any, Iterator, Union, Tuple
+
+
+# TODO(#1041): Use typing.NamedTuple once we require Python 3.
+class Platform(object):
```

Comment:
I would try to determine some quick canonical performance test just since I believe there is a lower memory footprint and possibly faster field access (unsure) when using `namedtuple` in python 2.

Reply:
I think field access might be ever so slightly faster.

For memory footprint, indeed. This is why I set `__slots__`. With `__slots__`, it shouldn't be very noticeable. (We decided in Pants that `__slots__` weren't worth the complexity; but I reason that Pex has different goals and constraints, such as minimizing its footprint. And this brings us closer to what `namedtuple` was doing.)

Reply:
Oh yes! Thank you for pointing that out.